### DISCUSSION TITLE
Minimizing Certain Statistics

### DISCUSSION THREAD

**Discussion by jbram22** [kernc/backtesting.py#370] [General]
I was wondering if there is a way to use the existing optimization class to minimize certain statistics? At the moment, I am only able to optimize my model in an effort to maximize certain statistics such as 'Final Equity', 'Winning Trade Rate', etc., but I think it will also be a useful exercise to optimize my model to minimize certain statistics such as 'Volatility', 'Avg. Drawdown', etc.

As a side note I love how easy & straightforward Backtest.py is!

**Comment by kernc** [comment]
You'll also love how simple custom optimization is: see https://github.com/kernc/backtesting.py/issues/234, https://github.com/kernc/backtesting.py/issues/256.

**Reply by FilJed** [reply]
Hi! As many of others, would love to thank you for such amasing repo. It is realy powerfull and easy-to-use. But all that functionality doesn't cover with documentation. There a lot of smal trick that is hidden in discussions.

Currently, while working on project of my own, I'm collectiong all this small tricks I found neat and usefull. May be that is a good idea to have repo's wiki with them? I would love to contribute gathering and maintaining it.

Sorry for texting you here.

**Reply by kernc** [reply]
What sort of tricks have you got in mind and how many do you have? I ask because the trick at hand (i.e. minimizing the objective) should be fairly easy to infer for anyone that [read the docs](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest.optimize):

> `maximize` is a string key from the [Backtest.run()](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest.run)-returned results series, or a function that accepts this series object and returns a number, the higher the better.

is fluent in Python and knows that one reverses the order of a set by negating the members.

**Reply by kernc** [reply]
FWIW; it was envisioned for major features and tricks to be showcased through [tutorials](https://kernc.github.io/backtesting.py/doc/backtesting/index.html#tutorials), which are [Jupyter notebooks](https://jupyter.org). Welcome to contribute. :smiley:
